### PR TITLE
spidershim: implement parts of Isolate::GetHeapStatistics

### LINKED ISSUE #114: Implement v8::Isolate::HeapStatistics
Looks like some of these can be implemented from JS::CollectRuntimeStats https://dxr.mozilla.org/mozilla-central/source/js/public/MemoryMetrics.h

### PR DESCRIPTION
**Pull request by brendandahl**
It may be possible to fill in a few more of these, but I'll need to get some more feedback from people who know spidermonkey's memory data better.

Fixes #114

### CODE REVIEW

File: deps/spidershim/src/v8isolate.cc [hunk 1]
```diff
@@ -313,4 +315,28 @@ void* Isolate::GetData(uint32_t slot) {
...
+    {}
+};
+
+void Isolate::GetHeapStatistics(HeapStatistics *heap_statistics) {
+  SimpleJSRuntimeStats rtStats(moz_malloc_size_of);
+  if (!JS::CollectRuntimeStats(Runtime(), &rtStats, nullptr, false)) {
+    return;
+  }
+  heap_statistics->total_heap_size_ = rtStats.gcHeapChunkTotal;
+  heap_statistics->used_heap_size_ = rtStats.gcHeapGCThings;
```
Comment (ehsan) [inline deps/spidershim/src/v8isolate.cc:365]: From what I read in the [header docs](https://dxr.mozilla.org/mozilla-central/source/js/public/MemoryMetrics.h#823), looks like what you want here is `rtStats.zTotals.sizeOfLiveGCThings() + rtStats.cTotals.sizeOfLiveGCThings()`.  But we can change this later...

Comment (ehsan) [inline deps/spidershim/src/v8isolate.cc:365]: Filed #117 as a follow-up.

**Comment by ehsan** [comment]
@brendandahl Can you please rebase this?  Thanks!

File: deps/spidershim/src/v8isolate.cc (hunk 1 above)
Comment (brendandahl) [inline deps/spidershim/src/v8isolate.cc:365]: I may be mis-reading the comment in the header docs but it seems to say:
`rtStats.gcHeapGCThings (in-use GC things) == rtStats.zTotals.sizeOfLiveGCThings() + rtStats.cTotals.sizeOfLiveGCThings()`

Comment (ehsan) [inline deps/spidershim/src/v8isolate.cc:365]: Right...  Yes.  I support your suggestion of asking someone who knows this stuff.  :-)

**Pull request merged**